animation_frame.h

```diff
@@ -0,0 +1,13 @@
+#ifndef ANIMATIONFRAME
+#define ANIMATIONFRAME
+#include "location.h"
+
+struct AnimationFrame {
+  SDL_Rect source;
+  Location offset;
+
+  AnimationFrame() : source({}), offset(0, 0) {}
+  AnimationFrame(SDL_Rect rect, Location location)
+      : source(rect), offset(location) {}
```

Comment:
I'm not convinced these are needed, but not a big deal to have them. Struct members are public by default, so I don't know that it adds much value to have a constructor.